lib/Runtime/Library/JavascriptLibrary.h

```diff
@@ -448,6 +448,7 @@ namespace Js
         bool inProfileMode;
         bool inDispatchProfileMode;
         bool arrayObjectHasUserDefinedSpecies;
+        bool arrayObjectHasUserDefinedIterator;
```

Comment:
With the simplification I mentioned below, this should no longer be needed.

I am also wondering if the species flag can be eliminated through simplification where it is used.